code/datums/diseases/advance/advance.dm

```diff
@@ -424,4 +424,9 @@ var/list/advance_cures = 	list(
 		src << "<a href='?_src_=vars;Vars=\ref[D]'>[D.name] - [D.holder]</a>"
 */
 
+/datum/symptom/heal/proc/Heal(mob/living/M, datum/disease/advance/A)
+	var/get_damage = sqrt(20-"stage_rate")*(2+rand())

```

Comment:
It uses the stage_rate code from above (hopefully) since I've had some difficulties finding the code for the total stage speed of viruses which has been an annoyance.